src/components/map/DesktopMap.jsx

```diff
@@ -240,26 +239,29 @@ class DesktopMap extends PureComponent {
                 plugin={PluginLeaflet}
                 onReady={() => { if (loading) setLoading(false); }}
               >
-                {layers.map((l, i) => (
-                  <Layer
-                    {...l}
-                    key={l.id}
-                    opacity={l.opacity}
-                    zIndex={1000 - i}
-                    {...l.params && { params: l.params }}
-                    {...l.sqlParams && { sqlParams: l.sqlParams }}
-                    {...l.decodeParams && { decodeParams: l.decodeParams }}
-                    {...l.interactionConfig && {
-                      interactivity: ['carto', 'cartodb'].includes(l.provider)
-                        ? (l.interactionConfig.output || []).map(o => o.column)
-                        : true
-                    }}
-                    events={{
-                      ...mapMode === 'analysis' && { click: (e) => { this.handlePoint(e, _map); } },
-                      ...mapMode === 'view' && { click: (e) => { this.handleClickMap(e); } }
-                    }}
-                  />
-                  ))}
+                {layers.map((layer, i) => {
+                  const l = { ...layer };
+                  return (
+                    <Layer
+                      {...l}
+                      key={l.id}
+                      opacity={l.opacity}
+                      zIndex={1000 - i}
+                      {...l.params && { params: l.params }}
+                      {...l.sqlParams && { sqlParams: l.sqlParams }}
+                      {...l.decodeParams && { decodeParams: l.decodeParams }}
+                      {...l.interactionConfig && {
+                        interactivity: ['carto', 'cartodb'].includes(l.provider)
+                          ? (l.interactionConfig.output || []).map(o => o.column)
+                          : true
+                      }}
+                      events={{
+                        ...mapMode === 'analysis' && { click: (e) => { this.handlePoint(e, _map); } },
+                        ...mapMode === 'view' && { click: (e) => { this.handleClickMap(e); } }
+                      }}
+                    />
+                  );
+                }).filter(e => e)}
```

Comment:
Is this filter still needed?

Reply:
@lbraun Probably not, I'll remove it and see if it is.  I think I was getting a null layer, but it was probably my tinkering that caused it.

Reply:
Filter was not needed